core/src/main/java/io/questdb/cairo/vm/MemoryCMARWImpl.java

```diff
@@ -245,11 +247,19 @@ public void setTruncateSize(long size) {
         jumpTo(size);
     }
 
-    public void sync(boolean async) {
-        if (pageAddress != 0 && ff.msync(pageAddress, size, async) == 0) {
-            return;
+    @Override
+    public void sync() {
+        if (pageAddress != 0 && commitMode != CommitMode.NOSYNC) {
+            if (ff.msync(pageAddress, size, commitMode == CommitMode.ASYNC) == 0) {
+                return;
+            }
+            LOG.error().$("could not msync [fd=").$(fd).$(", errno=").$(ff.errno()).$(']').$();
```

Comment:
Note. We'd better stop the database instance of msync errors as they're usually non-recoverable and stand for a critical problem.